### PR TITLE
call codecov and dscanner with circleci

### PR DESCRIPTION
**Pull request by wilzbach**
In reference to #106 - I still need to verify that it actually works.

### CODE REVIEW

**Comment by wilzbach** [comment]
> I still need to verify that it actually works.

[It does](https://circleci.com/gh/wilzbach/mir/41) :)

**Comment by 9il** [comment]
> I still need to verify that it actually works.
> It does :)

Where is report about coverage

**Comment by wilzbach** [comment]
> Where is report about coverage

It is waiting for AppVeyor and Travis to finish - this behavior didn't change. I don't know whether we can relax these constraints...

**Comment by wilzbach** [comment]
At least I see this on the CLI

```
bash <(curl -s https://codecov.io/bash)

  _____          _
 / ____|        | |
| |     ___   __| | ___  ___ _____   __
| |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
| |___| (_) | (_| |  __/ (_| (_) \ V /
 \_____\___/ \__,_|\___|\___\___/ \_/
                                e2e0ba7


(url) https://codecov.io
(root) .
==> Circle CI detected.
==> Running gcov (disable via -X gcov)
  -> gcov in .
==> Searching for coverage reports
**> Python coverage not found
==> Detecting git/mercurial file structure
==> Reading reports
  + ./..-..-..-tmp-dub_test_root-a949b99f-1f69-4ae0-bd59-9964f4eabfc0.lst bytes=1802
  + ./source-mir-combinatorics-package.lst bytes=49910
  + ./source-mir-las-sum.lst bytes=79118
  + ./source-mir-ndslice-internal.lst bytes=8621
  + ./source-mir-ndslice-iteration.lst bytes=42395
  + ./source-mir-ndslice-package.lst bytes=21312
  + ./source-mir-ndslice-selection.lst bytes=60109
  + ./source-mir-ndslice-slice.lst bytes=110156
(query) branch=circleci_dscanner&commit=1c5cd520a8bbf0f63973497a5642147d9d97c9a7&build=41.0&build_url=&slug=wilzbach/mir&service=circleci&pr=&job=
==> Uploading reports
    Pinging Codecov
    Uploading to S3

Reports queued to http://codecov.io/github/wilzbach/mir?ref=1c5cd520a8bbf0f63973497a5642147d9d97c9a7
```

**Comment by 9il** [comment]
Oh, so, it looks like codecove can combine reports from different builds(!!!)
I am really sorry, so than we need codecov on each arch, on Travis and AppVeyor, and only with recent DMD and recent LDC

**Comment by 9il** [comment]
So, we don't need codecove for circle CI

**Comment by wilzbach** [comment]
> I am really sorry, so than we need codecov on each arch, on Travis and AppVeyor, and only with recent DMD and recent LDC

this will require some `if` ...

> So, we don't need codecove for circle CI

okay but I guess we can keep `dscanner` here? :)

**Comment by 9il** [comment]
> okay but I guess we can keep dscanner here? :)

Yes)

**Comment by wilzbach** [comment]
Rebased :)

**Comment by wilzbach** [comment]
Thanks!

File: circle.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ dependencies:
     - curl -fsSL --retry 3 http://code.dlang.org/files/dub-${DUB}-linux-x86_64.tar.gz | tar -C ~/dmd2/linux/bin64 -zxf -
     - dmd --version
     - dub --version
+    - wget -O dscanner "http://releases.dlang.io/dscanner/0.3.0-git/dscanner"
+    - chmod +x dscanner
 test:
   override:
-    - dub test
+    - ./dscanner --config .dscanner.ini --styleCheck source 2> /dev/null
+    - dub test -b unittest-cov
```
Comment (9il) [inline circle.yml:18]: BTW, this should be just dub test

Comment (wilzbach) [inline circle.yml:18]: damn - I forgot that :/
Do you want to quickly change it?
(it doesn't hurt otherwise)

Comment (9il) [inline circle.yml:18]: This would be good to catch an error if any that can occurs without `unittest-cov`

Comment (wilzbach) [inline circle.yml:18]: ok - convinced ;-)

**Pull request merged**